### DISCUSSION TITLE
How can I verify authenticity of software I install via homebrew?

### DISCUSSION THREAD

**Discussion by masetime** [Homebrew/discussions#36] [Everyday usage]
I'm concerned about the possibility of malicious code being slipped into homebrew packages. I've searched around and not found an answer to this question.

Is there a method whereby I can verify that the packages being installed via homebrew are, indeed, genuine?

Perhaps something akin to the way I can verify I signed binary?

**Comment by SMillerDev** [comment]
That depends on what you're afraid of.

MITM attack: you could check the download checksum against the checksum defined in the formula.
Compromised build servers: you can build all software on your own device
